extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -0,0 +1,155 @@
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2017 Pentaho Corporation..  All rights reserved.
+ */
+package org.pentaho.platform.plugin.action;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.pentaho.platform.api.action.IActionInvokeStatus;
+import org.pentaho.platform.api.action.IActionInvoker;
+import org.pentaho.platform.api.engine.IPentahoSession;
+import org.pentaho.platform.api.engine.IPentahoSystemListener;
+import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.slf4j.MDC;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileFilter;
+
+public class ActionInvokerSystemListener implements IPentahoSystemListener {
+
+  private String environmentVariablesFilepath;
+  private static final Log logger = LogFactory.getLog( IPentahoSystemListener.class );
+  private static final String WORK_ITEM_FILE_EXTENSION = ".json";
+  private static final String DEFAULT_CONTENT_FOLDER = "system/default-content";
+
+  private Map<String, JSONObject> parseJsonFiles() throws IOException {
+    Map<String, JSONObject> dotJsons = new HashMap<>(  );
+    final String solutionPath = PentahoSystem.getApplicationContext().getSolutionPath( DEFAULT_CONTENT_FOLDER );
+    if ( solutionPath == null || solutionPath.isEmpty() ) {
+      logger.info( getClass().getName() + " Empty solution path" );
+    } else {
+      File[] jsonFiles = listFiles( new File( solutionPath ), WORK_ITEM_FILE_EXTENSION );
+      if ( jsonFiles != null ) {
+        for ( File jsonFile : jsonFiles ) {
+          JSONObject jsonObject = getJsonFromFile(  jsonFile  );
+          if ( jsonObject != null ) {
+            dotJsons.put( jsonFile.getAbsolutePath(), jsonObject );
+          }
+        }
+      }
+    }
+    return dotJsons;
+  }
+
+  private JSONObject getJsonFromFile( File file ) throws IOException {
+    FileReader fr = new FileReader( file );
+    try {
+      JSONParser jsonParser = new JSONParser();
+      JSONObject jsonObject = (JSONObject) jsonParser.parse( fr );
+      String jsonString = jsonObject.toJSONString();
+      fr.close();
+      return (JSONObject) jsonParser.parse( jsonString );
+    } catch ( ParseException e ) {
+      logger.error( "File: " + file.getAbsolutePath() + " is not valid json, and will not be processed" );
+      fr.close();
+      renameFile( file.getAbsolutePath() );
```

Comment:
it gets renamed later if this exception doesn't occur. renaming it at this point in the code would cause the chronos container to die which we only want to occur if we are at an end stage (exception or success).